### PR TITLE
Add TypeScript declarations

### PR DESCRIPTION
**Pull request by yoursunny**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+declare namespace CBOR {
+  export function encode(data: any, format?: "buffer"): Buffer;
```
Comment (thegecko) [inline index.d.ts:2]: Looks like the `buffer` form can also return a string:

https://github.com/ARMmbed/cbor-sync/blob/master/main.js#L518

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  export function encode(data: any, format: "hex"|"base64"): string;
```
Comment (thegecko) [inline index.d.ts:3]: And `hex` can return a `Buffer`:

https://github.com/ARMmbed/cbor-sync/blob/master/main.js#L592

**Review by thegecko — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 1 above)
Comment (yoursunny) [reply index.d.ts:2]: No, it cannot. When `format` is "buffer" or `undefined`,

https://github.com/ARMmbed/cbor-sync/blob/0b3dcad537433f4099562dd549681abdb988aa39/main.js#L533-L537
Here `BufferWriter` is constructed with `stringFormat=undefined`.

https://github.com/ARMmbed/cbor-sync/blob/0b3dcad537433f4099562dd549681abdb988aa39/main.js#L518
`result.toString` would not be executed.

File: index.d.ts (hunk 2 above)
Comment (yoursunny) [reply index.d.ts:3]: No, it cannot. The only construction of `HexWriter` is:
https://github.com/ARMmbed/cbor-sync/blob/0b3dcad537433f4099562dd549681abdb988aa39/main.js#L611-L613

It passes `finalFormat=undefined` to `HexWriter` constructor, and therefore `this.finalFormat` is "hex".
https://github.com/ARMmbed/cbor-sync/blob/0b3dcad537433f4099562dd549681abdb988aa39/main.js#L565

Consequently, `Buffer.from` would not be executed.
https://github.com/ARMmbed/cbor-sync/blob/0b3dcad537433f4099562dd549681abdb988aa39/main.js#L591-L593

**Comment by yoursunny** [comment]
I have explained my reasonings. Can you have a look again?

**Comment by thegecko** [comment]
Thanks, would you consider adding the other `api` members in a separate PR?
I can then create a new release.

- addReader()
- addWriter()
- config
- Reader
- Writer

**Pull request merged**